pkg/epp/requestcontrol/director.go

```diff
@@ -84,29 +98,72 @@ func (d *Director) HandleRequest(ctx context.Context, reqCtx *handlers.RequestCo
 		reqCtx.Request.Body["model"] = reqCtx.ResolvedTargetModel // Update target model in the body.
 	}
 
+	requestCriticality := v1alpha2.Standard
+	if modelObj.Spec.Criticality != nil {
+		requestCriticality = *modelObj.Spec.Criticality
+	}
+
+	// Prepare LLMRequest (needed for both saturation detection and Scheduler)
 	llmReq := &schedulingtypes.LLMRequest{
 		TargetModel: reqCtx.ResolvedTargetModel,
 		RequestId:   reqCtx.Request.Headers[requtil.RequestIdHeaderKey],
-		Critical:    modelObj.Spec.Criticality != nil && *modelObj.Spec.Criticality == v1alpha2.Critical,
+		Critical:    requestCriticality == v1alpha2.Critical,
 		Prompt:      prompt,
 		Headers:     reqCtx.Request.Headers,
 	}
-	logger.V(logutil.DEBUG).Info("LLM request assembled", "request", llmReq)
-	results, err := d.Dispatch(ctx, llmReq)
-	if err != nil {
-		return reqCtx, err
+	logger = logger.WithValues(
+		"model", reqCtx.Model,
+		"resolvedTargetModel", llmReq.TargetModel,
+		"criticality", requestCriticality,
+	)
+	ctx = log.IntoContext(ctx, logger)
+	logger.V(logutil.DEBUG).Info("LLM request assembled")
+
+	// --- 2. Saturation Check ---
+	logger.V(logutil.DEBUG).Info("Calling PreDispatch")
```

Comment:
please remove this and the ones below that log each function call, those will spam the logs

Reply:
removed and will be mindful of these moving forward